packages/dev/inspector/src/components/actionTabs/tabs/tools/metadataEditorComponent.tsx

```diff
@@ -0,0 +1,232 @@
+import { Inspector } from "inspector/inspector";
+import type { Scene } from "core/scene";
+import type { Nullable, Observable, Observer } from "@dev/core";
+import { ButtonLineComponent } from "shared-ui-components/lines/buttonLineComponent";
+import { TextInputLineComponent } from "shared-ui-components/lines/textInputLineComponent";
+import { CheckBoxLineComponent } from "shared-ui-components/lines/checkBoxLineComponent";
+
+import "./metadataEditor.scss";
+
+let currentEntity: any;
+const initialWindowSize = { width: 512, height: 512 };
+interface IMetadataEditorComponentProps {
+    scene: Scene;
+    globalState: GlobalState;
+}
+export const MetadataEditorComponent: React.FC<IMetadataEditorComponentProps> = (props: IMetadataEditorComponentProps) => {
+    const selectionUpdated = (entity: any) => {
+        currentEntity = entity;
+    };
+
+    const observableSelection: Observable<any> = props.globalState.onSelectionChangedObservable;
+    const observerMetaPopup: Nullable<Observer<any>> = observableSelection.add(selectionUpdated, -1, false, null, false);
+
+    const startMetadataEditorPopup = (): void => {
+        Inspector._CreatePersistentPopup(
+            {
+                props: {
+                    id: "metadata-editor",
+                    title: "Metadata Editor",
+                    onClose: () => observableSelection.remove(observerMetaPopup),
+                    size: initialWindowSize,
+                },
+                children: <MetadataEditorPopupComponent />,
+            },
+            document.body
+        );
+    };
+    return <ButtonLineComponent label="Open Metadata Editor" onClick={() => startMetadataEditorPopup()} />;
+};
+
+enum MetaDataTypes {
+    UNDEFINED = "undefined",
+    NULL = "null",
+    STRING = "string",
+    JSON = "JSON",
+}
+const MetadataEditorPopupComponent: React.FC = () => {
+    const [currentEntityUniqueId, setCurrentEntityUniqueId] = useState<string | null>(null);
+    const [currentEntityId, setCurrentEntityId] = useState("");
+    const [currentEntityName, setCurrentEntityName] = useState("");
+    const [currentEntityMetadata, setCurrentEntityMetadata] = useState<string>("");
+    const [metadataPropType, setMetadataPropType] = useState<string>(MetaDataTypes.UNDEFINED);
+    const [isValidJson, setIsValidJson] = useState(false);
+    const [prettyJson, setPrettyJson] = useState(true);
+    const [statusMessage, setStatusMessage] = useState<string | null>("ready to pick");
+
+    const stringifyEntityType = () => {
+        if (Object.prototype.hasOwnProperty.call(currentEntity, "metadata")) {
+            const meta = currentEntity.metadata;
+            if (isString(meta)) return MetaDataTypes.STRING;
+            return meta === null ? MetaDataTypes.NULL : MetaDataTypes.JSON;
+        }
+        return MetaDataTypes.UNDEFINED;
+    };
+
+    const isString = (input: any): boolean => {
+        return typeof input === "string" || input instanceof String;

```

Comment:
Is there a reason where we would need both typeof and instanceof here?

Reply:
```
typeof "" // 'string'
typeof new String("") // 'object'


"" instanceof String // true
new String() instanceof String // true
```

Reply:
Javascript being javascript... 😅